FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -1498,3 +1503,151 @@ void test_coverage_vTaskExitCriticalFromISR_isr_not_in_critical( void )
+ *     ...
+ *     taskEXIT_CRITICAL();
+ *     break;
+ * }
+ * @endcode
+ * ( pxTCB->xTaskRunState == taskTASK_NOT_RUNNING ) is true.
+ */
+void test_coverage_prvCheckTasksWaitingTermination_delete_not_running_task( void )
+{
+    TCB_t *pxTaskTCB = NULL;
+
+    /* Setup the variables and structure. */
+    uxDeletedTasksWaitingCleanUp = 1;
+    uxCurrentNumberOfTasks = 1;
+    vListInitialise( &xTasksWaitingTermination );
+
+    pxTaskTCB = pvPortMalloc( sizeof( TCB_t ) );
+    pxTaskTCB->pxStack = pvPortMalloc( configMINIMAL_STACK_SIZE );
+    pxTaskTCB->xTaskRunState = taskTASK_NOT_RUNNING ;
+    pxTaskTCB->xStateListItem.pvOwner = pxTaskTCB;
+    pxTaskTCB->xStateListItem.pxContainer = &xTasksWaitingTermination;
+    listINSERT_END( &xTasksWaitingTermination, &pxTaskTCB->xStateListItem );
+
+    /* Clear callback in commonSetUp. */
+    vFakePortEnterCriticalSection_StubWithCallback( NULL );
+    vFakePortExitCriticalSection_StubWithCallback( NULL );
+
+    /* Expectations. */
+    vFakePortEnterCriticalSection_Expect();
+    vFakePortExitCriticalSection_Expect();
+
+    /* API call. */
+    prvCheckTasksWaitingTermination();
+
+    /* Validation. */
+    TEST_ASSERT_EQUAL( uxCurrentNumberOfTasks, 0 );
+    TEST_ASSERT_EQUAL( uxDeletedTasksWaitingCleanUp, 0 );
+}
+
+/**
+ * @brief prvCheckTasksWaitingTermination - delete running task.
+ *
+ * A task to be deleted is still running. Nothing will be updated in this test case.
+ * The test shows it's result in the coverage report.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * if( pxTCB->xTaskRunState == taskTASK_NOT_RUNNING )
+ * {
+ *     ( void ) uxListRemove( &( pxTCB->xStateListItem ) );
+ *     --uxCurrentNumberOfTasks;
+ *     --uxDeletedTasksWaitingCleanUp;
+ * }
+ * else
+ * {
+ *     ...
+ *     taskEXIT_CRITICAL();
+ *     break;
+ * }
+ * @endcode
+ * ( pxTCB->xTaskRunState == taskTASK_NOT_RUNNING ) is false.
+ */
+void test_coverage_prvCheckTasksWaitingTermination_delete_running_task( void )
+{
+    TCB_t *pxTaskTCB = NULL;
+
+    /* Setup the variables and structure. */
+    uxDeletedTasksWaitingCleanUp = 1;
+    uxCurrentNumberOfTasks = 1;
+    vListInitialise( &xTasksWaitingTermination );
+
+    pxTaskTCB = pvPortMalloc( sizeof( TCB_t ) );
+    pxTaskTCB->pxStack = pvPortMalloc( configMINIMAL_STACK_SIZE );
+    pxTaskTCB->xTaskRunState = 0 ;
+    pxTaskTCB->xStateListItem.pvOwner = pxTaskTCB;
+    pxTaskTCB->xStateListItem.pxContainer = &xTasksWaitingTermination;
+    listINSERT_END( &xTasksWaitingTermination, &pxTaskTCB->xStateListItem );
+
+    /* Clear callback in commonSetUp. */
+    vFakePortEnterCriticalSection_StubWithCallback( NULL );
+    vFakePortExitCriticalSection_StubWithCallback( NULL );
+
+    /* Expectations. */
+    vFakePortEnterCriticalSection_Expect();
+    vFakePortExitCriticalSection_Expect();
+
+    /* API call. */
+    prvCheckTasksWaitingTermination();
+
+    /* Validation. */
+    /* If the task is not of taskTASK_NOT_RUNNING state, nothing will be updated.
```

Comment:
can we show that nothing has changed with some TEST_ASSERT_EQUAL? 
something before the call and the same thing after the call?

Reply:
Add assertion to ensure that the number of deleted task is the same.
Also verify the malloc count to ensure that the task is not deleted.